File: src/main/java/ru/vmsystems/template/infrastructure/persistence/ProcessRepository.java
package ru.vmsystems.template.infrastructure.persistence;


import org.springframework.data.repository.CrudRepository;
import ru.vmsystems.template.domain.model.CompanyEntity;
import ru.vmsystems.template.domain.model.ProcessEntity;

import java.util.List;

public interface ProcessRepository extends CrudRepository<ProcessEntity, Long> {

    List<ProcessEntity> getByCompanyId(Long companyId);

    List<ProcessEntity> getByCompany(CompanyEntity company);
}

File: src/main/java/ru/vmsystems/glasscloud/domain/process/ProcessRepository.kt
package ru.vmsystems.glasscloud.domain.process


import org.springframework.data.repository.CrudRepository
import java.util.*

interface ProcessRepository : CrudRepository<ProcessEntity, UUID> {
    fun getByCompanyId(companyId: UUID): List<ProcessEntity>
    fun getById(id: UUID): ProcessEntity?
}

interface ProcessTypeRepository : CrudRepository<ProcessTypeEntity, UUID> {
    fun getByCompanyId(companyId: UUID): List<ProcessTypeEntity>
    fun getById(id: UUID): ProcessTypeEntity?
}